router/throttler/throttler.go

```diff
@@ -0,0 +1,166 @@
+package throttler
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/EagleChen/restrictor"
+	"github.com/rudderlabs/rudder-server/config"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+)
+
+var (
+	parameters map[string]map[string]interface{}
```

Comment:
Let's create a new struct DestinationLimits to make this typed.

type DestinationThrottle struct {
			limit                     int,
			timeWindowInS             1,
      .....
} 

var allDestinationThrottles map[string]DestinationThrottle